### DISCUSSION TITLE
按需自动导入时vue3切换路由会卡顿且刷新，终端会提示加载组件css

### DISCUSSION THREAD

**Discussion by show110695** [element-plus/element-plus#16679] [Feature Request]
Feature type: **`Other`**

## Details

- Enhanced component name: `All`

## Description

按需自动导入时vue3切换路由会卡顿且刷新，终端会提示加载组件css，new dependencies optimized: element-plus/es/components/tag/style/css。是否考虑在官方文档里说明方案。目前我通过optimizeDeps.include解决，但担心会有问题。  
//  预加载element样式
  const optimizeDepsElementPlusIncludes = ['element-plus/es'];
  fs.readdirSync('node_modules/element-plus/es/components').forEach((dirname: string) => {
    fs.access(`node_modules/element-plus/es/components/${dirname}/style/index.mjs`, (err) => {
      if (!err) {
        optimizeDepsElementPlusIncludes.push(`element-plus/es/components/${dirname}/style/css`);
      }
    });
  });
optimizeDeps: {
      include: optimizeDepsElementPlusIncludes,
    }

### Additional comments

_(empty)_

**Comment by warmthsea** [comment]
https://github.com/element-plus/element-plus/blob/7bcaf29d06f23022d1eecfa75f391ad26ffe784b/play/vite.config.ts#L85C1-L87C7

这是 Vite 机制导致的，你可以这样处理或者手动 optimizeDeps 添加模块
不会有什么问题
